builtin/logical/pki/path_tidy_test.go

```diff
@@ -1443,16 +1455,30 @@ func TestSafetyBufferVsRevokedSafetyBuffer(t *testing.T) {
 	require.Nil(t, resp)
 
 	// Ensure the revoked certificate is not tidied before revoked_safety_buffer has passed
-	time.Sleep(2 * time.Second) // Check midway
+	time.Sleep(time.Until(revokedCert.NotAfter) + 3*time.Second)
```

Comment:
@fatima2003 On a fast enough system, 2s was before `revokedCert.NotAfter+safety_buffer`, so we needed to sleep until at least the `NotAfter` had been hit. I bumped `revoked_safety_buffer` to help accommodate system timing differences as well as a result.

Reply:
Ah yes since revoked certs won't be cleaned until after their expiration passes as well, Nice catch :D

Reply:
@fatima2003 Ah, it fails later, on line 1470 because I forgot to wait for auto-tidy to finish. I've fixed this and added the missing metric increment for tidying revocation in the `revoked/` check (which was also causing issues).

---

I don't think we want to do `time.Sleep(time.Until(revokedCert.NotAfter)/2)`, because we want to ensure `safety_buffer` does not delete the revoked certificate: this requires waiting until `NotAfter+safety_buffer` to validate. Waiting only `time.Until(revokedCert.NotAfter)/2` means we'll still be before the certificate expires, so neither check should remove it.

Reply:
Oh yes, my bad that is correct. 

---

The test times out after 30s when run locally: 

```
panic: test timed out after 30s
        running tests:
                TestSafetyBufferVsRevokedSafetyBuffer (30s)
```
<details>

```
goroutine 1522 [running]:
testing.(*M).startAlarm.func1()
        /usr/local/go/src/testing/testing.go:2373 +0x30c
created by time.goFunc
        /usr/local/go/src/time/sleep.go:215 +0x38

goroutine 1 [chan receive]:
testing.tRunner.func1()
        /usr/local/go/src/testing/testing.go:1651 +0x434
testing.tRunner(0x140008456c0, 0x1400090fc68)
        /usr/local/go/src/testing/testing.go:1696 +0x120
testing.runTests(0x1400061f2d8, {0x1048b51e0, 0xa2, 0xa2}, {0xeb00000000000000?, 0xeb2066b96adc8fd6?, 0x10498ba80?})
        /usr/local/go/src/testing/testing.go:2166 +0x3ac
testing.(*M).Run(0x14000885d60)
        /usr/local/go/src/testing/testing.go:2034 +0x588
main.main()
        _testmain.go:369 +0x90

goroutine 5 [select]:
go.opencensus.io/stats/view.(*worker).start(0x1400024d780)
        /Users/fatima/go/pkg/mod/go.opencensus.io@v0.24.0/stats/view/worker.go:292 +0x88
created by go.opencensus.io/stats/view.init.0 in goroutine 1
        /Users/fatima/go/pkg/mod/go.opencensus.io@v0.24.0/stats/view/worker.go:34 +0x98

goroutine 8 [sleep]:
time.Sleep(0x2acff0df8)
        /usr/local/go/src/runtime/time.go:285 +0xe0
github.com/openbao/openbao/builtin/logical/pki.TestSafetyBufferVsRevokedSafetyBuffer(0x14000845860)
        /Users/fatima/openbao/builtin/logical/pki/path_tidy_test.go:1483 +0x1064
testing.tRunner(0x14000845860, 0x103158e20)
        /usr/local/go/src/testing/testing.go:1690 +0xe4
created by testing.(*T).Run in goroutine 1
        /usr/local/go/src/testing/testing.go:1743 +0x314

goroutine 66 [syscall]:
os/signal.signal_recv()
        /usr/local/go/src/runtime/sigqueue.go:149 +0x2c
os/signal.loop()
        /usr/local/go/src/os/signal/signal_unix.go:23 +0x1c
created by os/signal.Notify.func1.1 in goroutine 8
        /usr/local/go/src/os/signal/signal.go:151 +0x28

goroutine 67 [select]:
github.com/armon/go-metrics.(*InmemSignal).run(...)
        /Users/fatima/go/pkg/mod/github.com/armon/go-metrics@v0.4.1/inmem_signal.go:64
created by github.com/armon/go-metrics.NewInmemSignal in goroutine 8
        /Users/fatima/go/pkg/mod/github.com/armon/go-metrics@v0.4.1/inmem_signal.go:38 +0x138

goroutine 68 [select]:
github.com/patrickmn/go-cache.(*janitor).Run(0x140001a7600, 0x140000e9c40)
        /Users/fatima/go/pkg/mod/github.com/patrickmn/go-cache@v2.1.0+incompatible/cache.go:1079 +0x78
created by github.com/patrickmn/go-cache.runJanitor in goroutine 8
        /Users/fatima/go/pkg/mod/github.com/patrickmn/go-cache@v2.1.0+incompatible/cache.go:1099 +0xe8

goroutine 69 [select]:
github.com/armon/go-metrics.(*InmemSignal).run(...)
        /Users/fatima/go/pkg/mod/github.com/armon/go-metrics@v0.4.1/inmem_signal.go:64
created by github.com/armon/go-metrics.NewInmemSignal in goroutine 8
        /Users/fatima/go/pkg/mod/github.com/armon/go-metrics@v0.4.1/inmem_signal.go:38 +0x138

goroutine 70 [select]:
github.com/patrickmn/go-cache.(*janitor).Run(0x14000062490, 0x140006f7800)
        /Users/fatima/go/pkg/mod/github.com/patrickmn/go-cache@v2.1.0+incompatible/cache.go:1079 +0x78
created by github.com/patrickmn/go-cache.runJanitor in goroutine 8
        /Users/fatima/go/pkg/mod/github.com/patrickmn/go-cache@v2.1.0+incompatible/cache.go:1099 +0xe8

goroutine 71 [select]:
github.com/armon/go-metrics.(*InmemSignal).run(...)
        /Users/fatima/go/pkg/mod/github.com/armon/go-metrics@v0.4.1/inmem_signal.go:64
created by github.com/armon/go-metrics.NewInmemSignal in goroutine 8
        /Users/fatima/go/pkg/mod/github.com/armon/go-metrics@v0.4.1/inmem_signal.go:38 +0x138

goroutine 72 [select]:
github.com/patrickmn/go-cache.(*janitor).Run(0x1400083c530, 0x14000528f40)
        /Users/fatima/go/pkg/mod/github.com/patrickmn/go-cache@v2.1.0+incompatible/cache.go:1079 +0x78
created by github.com/patrickmn/go-cache.runJanitor in goroutine 8
        /Users/fatima/go/pkg/mod/github.com/patrickmn/go-cache@v2.1.0+incompatible/cache.go:1099 +0xe8

goroutine 596 [select]:
google.golang.org/grpc/internal/transport.(*http2Client).keepalive(0x14000628908)
        /Users/fatima/go/pkg/mod/google.golang.org/grpc@v1.62.1/internal/transport/http2_client.go:1667 +0x100
created by google.golang.org/grpc/internal/transport.newHTTP2Client in goroutine 462
        /Users/fatima/go/pkg/mod/google.golang.org/grpc@v1.62.1/internal/transport/http2_client.go:390 +0x1458

goroutine 249 [sync.Cond.Wait]:
sync.runtime_notifyListWait(0x140001bb550, 0x0)
        /usr/local/go/src/runtime/sema.go:587 +0x154
sync.(*Cond).Wait(0x140001bb540)
        /usr/local/go/src/sync/cond.go:71 +0xcc
github.com/openbao/openbao/sdk/v2/physical/inmem.(*InmemLock).Lock.func1()
        /Users/fatima/openbao/sdk/physical/inmem/inmem_ha.go:87 +0xc8
created by github.com/openbao/openbao/sdk/v2/physical/inmem.(*InmemLock).Lock in goroutine 248
        /Users/fatima/openbao/sdk/physical/inmem/inmem_ha.go:82 +0x154

goroutine 282 [IO wait]:
internal/poll.runtime_pollWait(0x12ba79bf0, 0x72)
        /usr/local/go/src/runtime/netpoll.go:351 +0xa0
internal/poll.(*pollDesc).wait(0x1400024c180?, 0x1002c6650?, 0x0)
        /usr/local/go/src/internal/poll/fd_poll_runtime.go:84 +0x28
internal/poll.(*pollDesc).waitRead(...)
        /usr/local/go/src/internal/poll/fd_poll_runtime.go:89
internal/poll.(*FD).Accept(0x1400024c180)
        /usr/local/go/src/internal/poll/fd_unix.go:620 +0x24c
net.(*netFD).accept(0x1400024c180)
        /usr/local/go/src/net/fd_unix.go:172 +0x28
net.(*TCPListener).accept(0x14000aeef40)
        /usr/local/go/src/net/tcpsock_posix.go:159 +0x28
net.(*TCPListener).Accept(0x14000aeef40)
        /usr/local/go/src/net/tcpsock.go:372 +0x2c
crypto/tls.(*listener).Accept(0x14001003338)
        /usr/local/go/src/crypto/tls/tls.go:67 +0x30
github.com/openbao/openbao/vault/cluster.(*Listener).Run.func1(0x14001091730, {0x103184550, 0x14001003338})
        /Users/fatima/openbao/vault/cluster/cluster.go:328 +0x10c
created by github.com/openbao/openbao/vault/cluster.(*Listener).Run in goroutine 8
        /Users/fatima/openbao/vault/cluster/cluster.go:301 +0x27c

goroutine 626 [select]:
golang.org/x/net/http2.(*serverConn).serve(0x140014161c0)
        /Users/fatima/go/pkg/mod/golang.org/x/net@v0.26.0/http2/server.go:979 +0x6ec
golang.org/x/net/http2.(*Server).serveConn(0x14000a3ec60, {0x10319c5b8, 0x14000a9a008}, 0x14001480f70, 0x0)
        /Users/fatima/go/pkg/mod/golang.org/x/net@v0.26.0/http2/server.go:572 +0x97c
golang.org/x/net/http2.(*Server).ServeConn(...)
        /Users/fatima/go/pkg/mod/golang.org/x/net@v0.26.0/http2/server.go:436
github.com/openbao/openbao/vault.(*requestForwardingHandler).Handoff.func2()
        /Users/fatima/openbao/vault/request_forwarding.go:168 +0xdc
created by github.com/openbao/openbao/vault.(*requestForwardingHandler).Handoff in goroutine 282
        /Users/fatima/openbao/vault/request_forwarding.go:167 +0x16c

goroutine 233 [select]:
github.com/openbao/openbao/vault.(*Core).periodicCheckKeyUpgrades(0x14000b96008, {0x1031937c0, 0x1049b79e0}, 0x14000906230)
        /Users/fatima/openbao/vault/ha.go:868 +0xd8
github.com/openbao/openbao/vault.(*Core).runStandby.func3()
        /Users/fatima/openbao/vault/ha.go:413 +0x34
github.com/oklog/run.(*Group).Run.func1({0x140009000a8?, 0x140009000c0?})
        /Users/fatima/go/pkg/mod/github.com/oklog/run@v1.1.0/group.go:38 +0x34
created by github.com/oklog/run.(*Group).Run in goroutine 283
        /Users/fatima/go/pkg/mod/github.com/oklog/run@v1.1.0/group.go:37 +0x68

goroutine 643 [IO wait]:
internal/poll.runtime_pollWait(0x12ba793b0, 0x72)
        /usr/local/go/src/runtime/netpoll.go:351 +0xa0
internal/poll.(*pollDesc).wait(0x1400152a400?, 0x14001497400?, 0x0)
        /usr/local/go/src/internal/poll/fd_poll_runtime.go:84 +0x28
internal/poll.(*pollDesc).waitRead(...)
        /usr/local/go/src/internal/poll/fd_poll_runtime.go:89
internal/poll.(*FD).Read(0x1400152a400, {0x14001497400, 0x500, 0x500})
        /usr/local/go/src/internal/poll/fd_unix.go:165 +0x1fc
net.(*netFD).Read(0x1400152a400, {0x14001497400?, 0x14001497400?, 0x5?})
        /usr/local/go/src/net/fd_posix.go:55 +0x28
net.(*conn).Read(0x14001504050, {0x14001497400?, 0x14000a9a1c8?, 0x8?})
        /usr/local/go/src/net/net.go:189 +0x34
crypto/tls.(*atLeastReader).Read(0x14000f1e030, {0x14001497400?, 0x0?, 0x14000f1e030?})
        /usr/local/go/src/crypto/tls/conn.go:809 +0x40
bytes.(*Buffer).ReadFrom(0x14000a9a2b8, {0x10316bbe0, 0x14000f1e030})
        /usr/local/go/src/bytes/buffer.go:211 +0x90
crypto/tls.(*Conn).readFromUntil(0x14000a9a008, {0x104dae7e0, 0x14001504050}, 0x0?)
        /usr/local/go/src/crypto/tls/conn.go:831 +0xd0
crypto/tls.(*Conn).readRecordOrCCS(0x14000a9a008, 0x0)
        /usr/local/go/src/crypto/tls/conn.go:629 +0x35c
crypto/tls.(*Conn).readRecord(...)
        /usr/local/go/src/crypto/tls/conn.go:591
crypto/tls.(*Conn).Read(0x14000a9a008, {0x140002ff000, 0x9, 0x1007ca860?})
        /usr/local/go/src/crypto/tls/conn.go:1385 +0x164
io.ReadAtLeast({0x104daea60, 0x14000a9a008}, {0x140002ff000, 0x9, 0x9}, 0x9)
        /usr/local/go/src/io/io.go:335 +0xa0
io.ReadFull(...)
        /usr/local/go/src/io/io.go:354
golang.org/x/net/http2.readFrameHeader({0x140002ff000, 0x9, 0x140010c8f78?}, {0x104daea60?, 0x14000a9a008?})
        /Users/fatima/go/pkg/mod/golang.org/x/net@v0.26.0/http2/frame.go:237 +0x58
golang.org/x/net/http2.(*Framer).ReadFrame(0x140002fefc0)
        /Users/fatima/go/pkg/mod/golang.org/x/net@v0.26.0/http2/frame.go:501 +0x78
golang.org/x/net/http2.(*serverConn).readFrames(0x140014161c0)
        /Users/fatima/go/pkg/mod/golang.org/x/net@v0.26.0/http2/server.go:858 +0x98
created by golang.org/x/net/http2.(*serverConn).serve in goroutine 626
        /Users/fatima/go/pkg/mod/golang.org/x/net@v0.26.0/http2/server.go:971 +0x460

goroutine 232 [chan receive]:
github.com/openbao/openbao/vault.(*Core).runStandby.func1()
        /Users/fatima/openbao/vault/ha.go:404 +0x24
github.com/oklog/run.(*Group).Run.func1({0x14000aea020?, 0x10315a048?})
        /Users/fatima/go/pkg/mod/github.com/oklog/run@v1.1.0/group.go:38 +0x34
created by github.com/oklog/run.(*Group).Run in goroutine 283
        /Users/fatima/go/pkg/mod/github.com/oklog/run@v1.1.0/group.go:37 +0x68

goroutine 235 [select]:
github.com/openbao/openbao/vault.(*Core).metricsLoop(0x14000b96008, 0x14000906310)
        /Users/fatima/openbao/vault/core_metrics.go:50 +0x118
github.com/openbao/openbao/vault.(*Core).runStandby.func7()
        /Users/fatima/openbao/vault/ha.go:436 +0x24
github.com/oklog/run.(*Group).Run.func1({0x140009000f0?, 0x14000900108?})
        /Users/fatima/go/pkg/mod/github.com/oklog/run@v1.1.0/group.go:38 +0x34
created by github.com/oklog/run.(*Group).Run in goroutine 283
        /Users/fatima/go/pkg/mod/github.com/oklog/run@v1.1.0/group.go:37 +0x68

goroutine 248 [select]:
github.com/openbao/openbao/sdk/v2/physical/inmem.(*InmemLock).Lock(0x14000c2a100, 0x14000baebd0)
        /Users/fatima/openbao/sdk/physical/inmem/inmem_ha.go:107 +0x188
github.com/openbao/openbao/vault.(*Core).acquireLock(0x14000b97508, {0x1031858c0, 0x14000c2a100}, 0x14000baebd0)
        /Users/fatima/openbao/vault/ha.go:1053 +0x48
github.com/openbao/openbao/vault.(*Core).waitForLeadership(0x14000b97508, 0x14000988f9f?, 0x14000d97730, 0x14000baebd0)
        /Users/fatima/openbao/vault/ha.go:499 +0x248
github.com/openbao/openbao/vault.(*Core).runStandby.func9()
        /Users/fatima/openbao/vault/ha.go:448 +0x2c
github.com/oklog/run.(*Group).Run.func1({0x14000b8ea80?, 0x140008fe270?})
        /Users/fatima/go/pkg/mod/github.com/oklog/run@v1.1.0/group.go:38 +0x34
created by github.com/oklog/run.(*Group).Run in goroutine 470
        /Users/fatima/go/pkg/mod/github.com/oklog/run@v1.1.0/group.go:37 +0x68

goroutine 234 [select]:
github.com/openbao/openbao/vault.(*Core).periodicLeaderRefresh(0x14000b96008, 0x0, 0x140009062a0)
        /Users/fatima/openbao/vault/ha.go:809 +0x8c
github.com/openbao/openbao/vault.(*Core).runStandby.func5()
        /Users/fatima/openbao/vault/ha.go:425 +0x28
github.com/oklog/run.(*Group).Run.func1({0x1400108c1c0?, 0x140009000d8?})
        /Users/fatima/go/pkg/mod/github.com/oklog/run@v1.1.0/group.go:38 +0x34
created by github.com/oklog/run.(*Group).Run in goroutine 283
        /Users/fatima/go/pkg/mod/github.com/oklog/run@v1.1.0/group.go:37 +0x68

goroutine 284 [select]:
github.com/openbao/openbao/vault.(*Core).autoRotateBarrierLoop(0x14000b96008, {0x1031938d8, 0x14000a16320})
        /Users/fatima/openbao/vault/core.go:3155 +0x98
created by github.com/openbao/openbao/vault.standardUnsealStrategy.unseal in goroutine 236
        /Users/fatima/openbao/vault/core.go:2233 +0x178

goroutine 96 [select]:
github.com/openbao/openbao/vault.(*Core).setupCachedMFAResponseAuth.func1()
        /Users/fatima/openbao/vault/core.go:3243 +0x98
created by github.com/openbao/openbao/vault.(*Core).setupCachedMFAResponseAuth in goroutine 8
        /Users/fatima/openbao/vault/core.go:3240 +0x180

goroutine 236 [select]:
github.com/openbao/openbao/vault.(*Core).waitForLeadership(0x14000b96008, 0x14000a17310?, 0x14001091810, 0x14000906380)
        /Users/fatima/openbao/vault/ha.go:652 +0xbbc
github.com/openbao/openbao/vault.(*Core).runStandby.func9()
        /Users/fatima/openbao/vault/ha.go:448 +0x2c
github.com/oklog/run.(*Group).Run.func1({0x14000ae0240?, 0x14000900120?})
        /Users/fatima/go/pkg/mod/github.com/oklog/run@v1.1.0/group.go:38 +0x34
created by github.com/oklog/run.(*Group).Run in goroutine 283
        /Users/fatima/go/pkg/mod/github.com/oklog/run@v1.1.0/group.go:37 +0x68

goroutine 672 [chan receive]:
github.com/gammazero/workerpool.worker(0x0?, 0x14000b6ea10, 0x14000c02030)
        /Users/fatima/go/pkg/mod/github.com/gammazero/workerpool@v1.1.3/workerpool.go:238 +0x40
created by github.com/gammazero/workerpool.(*WorkerPool).dispatch in goroutine 291
        /Users/fatima/go/pkg/mod/github.com/gammazero/workerpool@v1.1.3/workerpool.go:197 +0x214

goroutine 283 [chan receive]:
github.com/oklog/run.(*Group).Run(0x14000900090)
        /Users/fatima/go/pkg/mod/github.com/oklog/run@v1.1.0/group.go:43 +0x140
github.com/openbao/openbao/vault.(*Core).runStandby(0x14000b96008, 0x0?, 0x14001091810, 0x14001091880)
        /Users/fatima/openbao/vault/ha.go:457 +0x778
created by github.com/openbao/openbao/vault.(*Core).unsealInternal in goroutine 8
        /Users/fatima/openbao/vault/core.go:1867 +0x1f8

goroutine 172 [select]:
github.com/openbao/openbao/builtin/logical/pki.(*ACMEChallengeEngine)._run(0x14001608240, 0x140007b8140, 0x1400093e4d0)
        /Users/fatima/openbao/builtin/logical/pki/acme_challenge_engine.go:126 +0xf4
github.com/openbao/openbao/builtin/logical/pki.(*ACMEChallengeEngine).Run(0x14001608240, 0x140007b8140, 0x1400093e4d0, 0x14001608b10)
        /Users/fatima/openbao/builtin/logical/pki/acme_challenge_engine.go:105 +0x20c
created by github.com/openbao/openbao/builtin/logical/pki.(*acmeState).Initialize in goroutine 536
        /Users/fatima/openbao/builtin/logical/pki/acme_state.go:88 +0x1d4

goroutine 515 [select]:
github.com/openbao/openbao/vault.(*forwardingClient).startHeartbeat.func1()
        /Users/fatima/openbao/vault/request_forwarding_rpc.go:171 +0x1dc
created by github.com/openbao/openbao/vault.(*forwardingClient).startHeartbeat in goroutine 8
        /Users/fatima/openbao/vault/request_forwarding_rpc.go:119 +0x5c

goroutine 287 [select]:
github.com/patrickmn/go-cache.(*janitor).Run(0x14000a0afe0, 0x140006f7f40)
        /Users/fatima/go/pkg/mod/github.com/patrickmn/go-cache@v2.1.0+incompatible/cache.go:1079 +0x78
created by github.com/patrickmn/go-cache.runJanitor in goroutine 236
        /Users/fatima/go/pkg/mod/github.com/patrickmn/go-cache@v2.1.0+incompatible/cache.go:1099 +0xe8

goroutine 290 [select]:
github.com/openbao/openbao/vault.(*Core).updateLockedUserEntries.func1()
        /Users/fatima/openbao/vault/core.go:3274 +0x8c
created by github.com/openbao/openbao/vault.(*Core).updateLockedUserEntries in goroutine 236
        /Users/fatima/openbao/vault/core.go:3271 +0x178

goroutine 291 [select]:
github.com/gammazero/workerpool.(*WorkerPool).dispatch(0x14000a23280)
        /Users/fatima/go/pkg/mod/github.com/gammazero/workerpool@v1.1.3/workerpool.go:185 +0xec
created by github.com/gammazero/workerpool.New in goroutine 236
        /Users/fatima/go/pkg/mod/github.com/gammazero/workerpool@v1.1.3/workerpool.go:37 +0x124

goroutine 292 [select]:
github.com/openbao/openbao/vault.(*RollbackManager).run(0x14000b6c160)
        /Users/fatima/openbao/vault/rollback.go:152 +0x100
created by github.com/openbao/openbao/vault.(*RollbackManager).Start in goroutine 236
        /Users/fatima/openbao/vault/rollback.go:115 +0x5c

goroutine 293 [select]:
github.com/openbao/openbao/helper/fairshare.(*worker).start.func1()
        /Users/fatima/openbao/helper/fairshare/workerpool.go:55 +0x74
created by github.com/openbao/openbao/helper/fairshare.(*worker).start in goroutine 236
        /Users/fatima/openbao/helper/fairshare/workerpool.go:53 +0x6c

goroutine 294 [select]:
github.com/openbao/openbao/helper/fairshare.(*worker).start.func1()
        /Users/fatima/openbao/helper/fairshare/workerpool.go:55 +0x74
created by github.com/openbao/openbao/helper/fairshare.(*worker).start in goroutine 236
        /Users/fatima/openbao/helper/fairshare/workerpool.go:53 +0x6c

goroutine 295 [select]:
github.com/openbao/openbao/helper/fairshare.(*worker).start.func1()
        /Users/fatima/openbao/helper/fairshare/workerpool.go:55 +0x74
created by github.com/openbao/openbao/helper/fairshare.(*worker).start in goroutine 236
        /Users/fatima/openbao/helper/fairshare/workerpool.go:53 +0x6c

goroutine 296 [select]:
github.com/openbao/openbao/helper/fairshare.(*worker).start.func1()
        /Users/fatima/openbao/helper/fairshare/workerpool.go:55 +0x74
created by github.com/openbao/openbao/helper/fairshare.(*worker).start in goroutine 236
        /Users/fatima/openbao/helper/fairshare/workerpool.go:53 +0x6c

goroutine 297 [select]:
github.com/openbao/openbao/helper/fairshare.(*worker).start.func1()
        /Users/fatima/openbao/helper/fairshare/workerpool.go:55 +0x74
created by github.com/openbao/openbao/helper/fairshare.(*worker).start in goroutine 236
        /Users/fatima/openbao/helper/fairshare/workerpool.go:53 +0x6c

goroutine 298 [select]:
github.com/openbao/openbao/helper/fairshare.(*worker).start.func1()
        /Users/fatima/openbao/helper/fairshare/workerpool.go:55 +0x74
created by github.com/openbao/openbao/helper/fairshare.(*worker).start in goroutine 236
        /Users/fatima/openbao/helper/fairshare/workerpool.go:53 +0x6c

goroutine 299 [select]:
github.com/openbao/openbao/helper/fairshare.(*worker).start.func1()
        /Users/fatima/openbao/helper/fairshare/workerpool.go:55 +0x74
created by github.com/openbao/openbao/helper/fairshare.(*worker).start in goroutine 236
        /Users/fatima/openbao/helper/fairshare/workerpool.go:53 +0x6c

goroutine 300 [select]:
github.com/openbao/openbao/helper/fairshare.(*worker).start.func1()
        /Users/fatima/openbao/helper/fairshare/workerpool.go:55 +0x74
created by github.com/openbao/openbao/helper/fairshare.(*worker).start in goroutine 236
        /Users/fatima/openbao/helper/fairshare/workerpool.go:53 +0x6c

goroutine 301 [select]:
github.com/openbao/openbao/helper/fairshare.(*worker).start.func1()
        /Users/fatima/openbao/helper/fairshare/workerpool.go:55 +0x74
created by github.com/openbao/openbao/helper/fairshare.(*worker).start in goroutine 236
        /Users/fatima/openbao/helper/fairshare/workerpool.go:53 +0x6c

goroutine 302 [select]:
github.com/openbao/openbao/helper/fairshare.(*worker).start.func1()
        /Users/fatima/openbao/helper/fairshare/workerpool.go:55 +0x74
created by github.com/openbao/openbao/helper/fairshare.(*worker).start in goroutine 236
        /Users/fatima/openbao/helper/fairshare/workerpool.go:53 +0x6c

goroutine 303 [select]:
github.com/openbao/openbao/helper/fairshare.(*JobManager).assignWork.func1()
        /Users/fatima/openbao/helper/fairshare/jobmanager.go:302 +0xc4
created by github.com/openbao/openbao/helper/fairshare.(*JobManager).assignWork in goroutine 236
        /Users/fatima/openbao/helper/fairshare/jobmanager.go:270 +0x70

goroutine 304 [select]:
github.com/openbao/openbao/vault.(*ExpirationManager).uniquePoliciesGc(0x14000a923c0)
        /Users/fatima/openbao/vault/expiration.go:1754 +0x70
created by github.com/openbao/openbao/vault.NewExpirationManager in goroutine 236
        /Users/fatima/openbao/vault/expiration.go:385 +0x544

goroutine 243 [select]:
github.com/patrickmn/go-cache.(*janitor).Run(0x14000c26180, 0x14000c2a080)
        /Users/fatima/go/pkg/mod/github.com/patrickmn/go-cache@v2.1.0+incompatible/cache.go:1079 +0x78
created by github.com/patrickmn/go-cache.runJanitor in goroutine 236
        /Users/fatima/go/pkg/mod/github.com/patrickmn/go-cache@v2.1.0+incompatible/cache.go:1099 +0xe8

goroutine 306 [select]:
github.com/openbao/openbao/vault.(*Core).setupExpiration.func2()
        /Users/fatima/openbao/vault/expiration.go:429 +0xb0
created by github.com/openbao/openbao/vault.(*Core).setupExpiration in goroutine 236
        /Users/fatima/openbao/vault/expiration.go:426 +0x31c

goroutine 247 [select]:
github.com/openbao/openbao/vault.(*Core).metricsLoop(0x14000b97508, 0x14000baeb60)
        /Users/fatima/openbao/vault/core_metrics.go:50 +0x118
github.com/openbao/openbao/vault.(*Core).runStandby.func7()
        /Users/fatima/openbao/vault/ha.go:436 +0x24
github.com/oklog/run.(*Group).Run.func1({0x140008fe228?, 0x140008fe240?})
        /Users/fatima/go/pkg/mod/github.com/oklog/run@v1.1.0/group.go:38 +0x34
created by github.com/oklog/run.(*Group).Run in goroutine 470
        /Users/fatima/go/pkg/mod/github.com/oklog/run@v1.1.0/group.go:37 +0x68

goroutine 471 [IO wait]:
internal/poll.runtime_pollWait(0x12ba799e0, 0x72)
        /usr/local/go/src/runtime/netpoll.go:351 +0xa0
internal/poll.(*pollDesc).wait(0x14000dd1b80?, 0x1002c6650?, 0x0)
        /usr/local/go/src/internal/poll/fd_poll_runtime.go:84 +0x28
internal/poll.(*pollDesc).waitRead(...)
        /usr/local/go/src/internal/poll/fd_poll_runtime.go:89
internal/poll.(*FD).Accept(0x14000dd1b80)
        /usr/local/go/src/internal/poll/fd_unix.go:620 +0x24c
net.(*netFD).accept(0x14000dd1b80)
        /usr/local/go/src/net/fd_unix.go:172 +0x28
net.(*TCPListener).accept(0x14000da4ec0)
        /usr/local/go/src/net/tcpsock_posix.go:159 +0x28
net.(*TCPListener).Accept(0x14000da4ec0)
        /usr/local/go/src/net/tcpsock.go:372 +0x2c
crypto/tls.(*listener).Accept(0x140009808e8)
        /usr/local/go/src/crypto/tls/tls.go:67 +0x30
github.com/openbao/openbao/vault/cluster.(*Listener).Run.func1(0x14000d97810, {0x103184550, 0x140009808e8})
        /Users/fatima/openbao/vault/cluster/cluster.go:328 +0x10c
created by github.com/openbao/openbao/vault/cluster.(*Listener).Run in goroutine 8
        /Users/fatima/openbao/vault/cluster/cluster.go:301 +0x27c

goroutine 549 [IO wait]:
internal/poll.runtime_pollWait(0x12ba797d0, 0x72)
        /usr/local/go/src/runtime/netpoll.go:351 +0xa0
internal/poll.(*pollDesc).wait(0x14000a23a80?, 0x1400050e900?, 0x0)
        /usr/local/go/src/internal/poll/fd_poll_runtime.go:84 +0x28
internal/poll.(*pollDesc).waitRead(...)
        /usr/local/go/src/internal/poll/fd_poll_runtime.go:89
internal/poll.(*FD).Read(0x14000a23a80, {0x1400050e900, 0x480, 0x480})
        /usr/local/go/src/internal/poll/fd_unix.go:165 +0x1fc
net.(*netFD).Read(0x14000a23a80, {0x1400050e900?, 0x1400092f978?, 0x1002d88d8?})
        /usr/local/go/src/net/fd_posix.go:55 +0x28
net.(*conn).Read(0x14000c00f40, {0x1400050e900?, 0x1400092f9f8?, 0x1006a139c?})
        /usr/local/go/src/net/net.go:189 +0x34
crypto/tls.(*atLeastReader).Read(0x14001514108, {0x1400050e900?, 0x0?, 0x14001514108?})
        /usr/local/go/src/crypto/tls/conn.go:809 +0x40
bytes.(*Buffer).ReadFrom(0x14001279eb8, {0x10316bbe0, 0x14001514108})
        /usr/local/go/src/bytes/buffer.go:211 +0x90
crypto/tls.(*Conn).readFromUntil(0x14001279c08, {0x104dae7e0, 0x14000c00f40}, 0x0?)
        /usr/local/go/src/crypto/tls/conn.go:831 +0xd0
crypto/tls.(*Conn).readRecordOrCCS(0x14001279c08, 0x0)
        /usr/local/go/src/crypto/tls/conn.go:629 +0x35c
crypto/tls.(*Conn).readRecord(...)
        /usr/local/go/src/crypto/tls/conn.go:591
crypto/tls.(*Conn).Read(0x14001279c08, {0x14001538120, 0x9, 0x1006a68c0?})
        /usr/local/go/src/crypto/tls/conn.go:1385 +0x164
io.ReadAtLeast({0x104daea60, 0x14001279c08}, {0x14001538120, 0x9, 0x9}, 0x9)
        /usr/local/go/src/io/io.go:335 +0xa0
io.ReadFull(...)
        /usr/local/go/src/io/io.go:354
net/http.http2readFrameHeader({0x14001538120, 0x9, 0x14000fcfec0?}, {0x104daea60?, 0x14001279c08?})
        /usr/local/go/src/net/http/h2_bundle.go:1642 +0x58
net/http.(*http2Framer).ReadFrame(0x140015380e0)
        /usr/local/go/src/net/http/h2_bundle.go:1909 +0x78
net/http.(*http2serverConn).readFrames(0x14001544000)
        /usr/local/go/src/net/http/h2_bundle.go:4691 +0x98
created by net/http.(*http2serverConn).serve in goroutine 534
        /usr/local/go/src/net/http/h2_bundle.go:4804 +0x460

goroutine 518 [select]:
google.golang.org/grpc/internal/grpcsync.(*CallbackSerializer).run(0x14001097f40, {0x1031938d8, 0x1400109bc70})
        /Users/fatima/go/pkg/mod/google.golang.org/grpc@v1.62.1/internal/grpcsync/callback_serializer.go:76 +0xc4
created by google.golang.org/grpc/internal/grpcsync.NewCallbackSerializer in goroutine 8
        /Users/fatima/go/pkg/mod/google.golang.org/grpc@v1.62.1/internal/grpcsync/callback_serializer.go:52 +0x118

goroutine 517 [select]:
google.golang.org/grpc/internal/grpcsync.(*CallbackSerializer).run(0x14001097f10, {0x1031938d8, 0x1400109bc20})
        /Users/fatima/go/pkg/mod/google.golang.org/grpc@v1.62.1/internal/grpcsync/callback_serializer.go:76 +0xc4
created by google.golang.org/grpc/internal/grpcsync.NewCallbackSerializer in goroutine 8
        /Users/fatima/go/pkg/mod/google.golang.org/grpc@v1.62.1/internal/grpcsync/callback_serializer.go:52 +0x118

goroutine 245 [select]:
github.com/openbao/openbao/vault.(*Core).periodicCheckKeyUpgrades(0x14000b97508, {0x1031937c0, 0x1049b79e0}, 0x14000baea80)
        /Users/fatima/openbao/vault/ha.go:868 +0xd8
github.com/openbao/openbao/vault.(*Core).runStandby.func3()
        /Users/fatima/openbao/vault/ha.go:413 +0x34
github.com/oklog/run.(*Group).Run.func1({0x140008fe1e0?, 0x140008fe1f8?})
        /Users/fatima/go/pkg/mod/github.com/oklog/run@v1.1.0/group.go:38 +0x34
created by github.com/oklog/run.(*Group).Run in goroutine 470
        /Users/fatima/go/pkg/mod/github.com/oklog/run@v1.1.0/group.go:37 +0x68

goroutine 563 [select]:
golang.org/x/net/http2.(*serverConn).serve(0x140015441c0)
        /Users/fatima/go/pkg/mod/golang.org/x/net@v0.26.0/http2/server.go:979 +0x6ec
golang.org/x/net/http2.(*Server).serveConn(0x14000a3ec60, {0x10319c5b8, 0x14000c6a008}, 0x14001484f70, 0x0)
        /Users/fatima/go/pkg/mod/golang.org/x/net@v0.26.0/http2/server.go:572 +0x97c
golang.org/x/net/http2.(*Server).ServeConn(...)
        /Users/fatima/go/pkg/mod/golang.org/x/net@v0.26.0/http2/server.go:436
github.com/openbao/openbao/vault.(*requestForwardingHandler).Handoff.func2()
        /Users/fatima/openbao/vault/request_forwarding.go:168 +0xdc
created by github.com/openbao/openbao/vault.(*requestForwardingHandler).Handoff in goroutine 282
        /Users/fatima/openbao/vault/request_forwarding.go:167 +0x16c

goroutine 516 [select]:
google.golang.org/grpc/internal/grpcsync.(*CallbackSerializer).run(0x14001097ee0, {0x1031938d8, 0x1400109bbd0})
        /Users/fatima/go/pkg/mod/google.golang.org/grpc@v1.62.1/internal/grpcsync/callback_serializer.go:76 +0xc4
created by google.golang.org/grpc/internal/grpcsync.NewCallbackSerializer in goroutine 8
        /Users/fatima/go/pkg/mod/google.golang.org/grpc@v1.62.1/internal/grpcsync/callback_serializer.go:52 +0x118

goroutine 111 [select]:
google.golang.org/grpc/internal/grpcsync.(*CallbackSerializer).run(0x140010975b0, {0x1031938d8, 0x1400109b900})
        /Users/fatima/go/pkg/mod/google.golang.org/grpc@v1.62.1/internal/grpcsync/callback_serializer.go:76 +0xc4
created by google.golang.org/grpc/internal/grpcsync.NewCallbackSerializer in goroutine 8
        /Users/fatima/go/pkg/mod/google.golang.org/grpc@v1.62.1/internal/grpcsync/callback_serializer.go:52 +0x118

goroutine 480 [chan receive]:
github.com/gammazero/workerpool.worker(0x14000b961d8?, 0x14000b6ea10, 0x14000c02030)
        /Users/fatima/go/pkg/mod/github.com/gammazero/workerpool@v1.1.3/workerpool.go:238 +0x40
created by github.com/gammazero/workerpool.(*WorkerPool).dispatch in goroutine 291
        /Users/fatima/go/pkg/mod/github.com/gammazero/workerpool@v1.1.3/workerpool.go:197 +0x214

goroutine 103 [chan receive]:
github.com/gammazero/workerpool.worker(0x0?, 0x14000b6ea10, 0x14000c02030)
        /Users/fatima/go/pkg/mod/github.com/gammazero/workerpool@v1.1.3/workerpool.go:238 +0x40
created by github.com/gammazero/workerpool.(*WorkerPool).dispatch in goroutine 291
        /Users/fatima/go/pkg/mod/github.com/gammazero/workerpool@v1.1.3/workerpool.go:197 +0x214

goroutine 513 [select]:
github.com/openbao/openbao/vault.(*requestForwardingHandler).Handoff.func1()
        /Users/fatima/openbao/vault/request_forwarding.go:158 +0x70
created by github.com/openbao/openbao/vault.(*requestForwardingHandler).Handoff in goroutine 282
        /Users/fatima/openbao/vault/request_forwarding.go:157 +0xfc

goroutine 26 [select]:
google.golang.org/grpc/internal/transport.(*http2Client).keepalive(0x14000628248)
        /Users/fatima/go/pkg/mod/google.golang.org/grpc@v1.62.1/internal/transport/http2_client.go:1667 +0x100
created by google.golang.org/grpc/internal/transport.newHTTP2Client in goroutine 514
        /Users/fatima/go/pkg/mod/google.golang.org/grpc@v1.62.1/internal/transport/http2_client.go:390 +0x1458

goroutine 27 [IO wait]:
internal/poll.runtime_pollWait(0x12ba796c8, 0x72)
        /usr/local/go/src/runtime/netpoll.go:351 +0xa0
internal/poll.(*pollDesc).wait(0x14001019c80?, 0x14001596000?, 0x0)
        /usr/local/go/src/internal/poll/fd_poll_runtime.go:84 +0x28
internal/poll.(*pollDesc).waitRead(...)
        /usr/local/go/src/internal/poll/fd_poll_runtime.go:89
internal/poll.(*FD).Read(0x14001019c80, {0x14001596000, 0x600, 0x600})
        /usr/local/go/src/internal/poll/fd_unix.go:165 +0x1fc
net.(*netFD).Read(0x14001019c80, {0x14001596000?, 0x14001596000?, 0x5?})
        /usr/local/go/src/net/fd_posix.go:55 +0x28
net.(*conn).Read(0x14000a24098, {0x14001596000?, 0x14000a741c8?, 0x8?})
        /usr/local/go/src/net/net.go:189 +0x34
crypto/tls.(*atLeastReader).Read(0x140001ada28, {0x14001596000?, 0x0?, 0x140001ada28?})
        /usr/local/go/src/crypto/tls/conn.go:809 +0x40
bytes.(*Buffer).ReadFrom(0x14000a742b8, {0x10316bbe0, 0x140001ada28})
        /usr/local/go/src/bytes/buffer.go:211 +0x90
crypto/tls.(*Conn).readFromUntil(0x14000a74008, {0x104dae7e0, 0x14000a24098}, 0x0?)
        /usr/local/go/src/crypto/tls/conn.go:831 +0xd0
crypto/tls.(*Conn).readRecordOrCCS(0x14000a74008, 0x0)
        /usr/local/go/src/crypto/tls/conn.go:629 +0x35c
crypto/tls.(*Conn).readRecord(...)
        /usr/local/go/src/crypto/tls/conn.go:591
crypto/tls.(*Conn).Read(0x14000a74008, {0x140009c4000, 0x8000, 0x1002c588c?})
        /usr/local/go/src/crypto/tls/conn.go:1385 +0x164
bufio.(*Reader).Read(0x14000d819e0, {0x14000db8040, 0x9, 0x1002e3460?})
        /usr/local/go/src/bufio/bufio.go:241 +0x1b4
io.ReadAtLeast({0x10316b800, 0x14000d819e0}, {0x14000db8040, 0x9, 0x9}, 0x9)
        /usr/local/go/src/io/io.go:335 +0xa0
io.ReadFull(...)
        /usr/local/go/src/io/io.go:354
golang.org/x/net/http2.readFrameHeader({0x14000db8040, 0x9, 0x689aea969?}, {0x10316b800?, 0x14000d819e0?})
        /Users/fatima/go/pkg/mod/golang.org/x/net@v0.26.0/http2/frame.go:237 +0x58
golang.org/x/net/http2.(*Framer).ReadFrame(0x14000db8000)
        /Users/fatima/go/pkg/mod/golang.org/x/net@v0.26.0/http2/frame.go:501 +0x78
google.golang.org/grpc/internal/transport.(*http2Client).reader(0x14000628248, 0x1400059a620)
        /Users/fatima/go/pkg/mod/google.golang.org/grpc@v1.62.1/internal/transport/http2_client.go:1593 +0x1a4
created by google.golang.org/grpc/internal/transport.newHTTP2Client in goroutine 514
        /Users/fatima/go/pkg/mod/google.golang.org/grpc@v1.62.1/internal/transport/http2_client.go:400 +0x14bc

goroutine 28 [select]:
google.golang.org/grpc/internal/transport.(*controlBuffer).get(0x14000ae4140, 0x1)
        /Users/fatima/go/pkg/mod/google.golang.org/grpc@v1.62.1/internal/transport/controlbuf.go:418 +0x14c
google.golang.org/grpc/internal/transport.(*loopyWriter).run(0x1400059a770)
        /Users/fatima/go/pkg/mod/google.golang.org/grpc@v1.62.1/internal/transport/controlbuf.go:551 +0x74
google.golang.org/grpc/internal/transport.newHTTP2Client.func6()
        /Users/fatima/go/pkg/mod/google.golang.org/grpc@v1.62.1/internal/transport/http2_client.go:454 +0x70
created by google.golang.org/grpc/internal/transport.newHTTP2Client in goroutine 514
        /Users/fatima/go/pkg/mod/google.golang.org/grpc@v1.62.1/internal/transport/http2_client.go:452 +0x190c

goroutine 597 [IO wait]:
internal/poll.runtime_pollWait(0x12ba795c0, 0x72)
        /usr/local/go/src/runtime/netpoll.go:351 +0xa0
internal/poll.(*pollDesc).wait(0x14000fd2680?, 0x14000dba000?, 0x0)
        /usr/local/go/src/internal/poll/fd_poll_runtime.go:84 +0x28
internal/poll.(*pollDesc).waitRead(...)
        /usr/local/go/src/internal/poll/fd_poll_runtime.go:89
internal/poll.(*FD).Read(0x14000fd2680, {0x14000dba000, 0x600, 0x600})
        /usr/local/go/src/internal/poll/fd_unix.go:165 +0x1fc
net.(*netFD).Read(0x14000fd2680, {0x14000dba000?, 0x14000dba000?, 0x5?})
        /usr/local/go/src/net/fd_posix.go:55 +0x28
net.(*conn).Read(0x14001504000, {0x14000dba000?, 0x140015061c8?, 0x8?})
        /usr/local/go/src/net/net.go:189 +0x34
crypto/tls.(*atLeastReader).Read(0x14001514030, {0x14000dba000?, 0x0?, 0x14001514030?})
        /usr/local/go/src/crypto/tls/conn.go:809 +0x40
bytes.(*Buffer).ReadFrom(0x140015062b8, {0x10316bbe0, 0x14001514030})
        /usr/local/go/src/bytes/buffer.go:211 +0x90
crypto/tls.(*Conn).readFromUntil(0x14001506008, {0x104dae7e0, 0x14001504000}, 0x0?)
        /usr/local/go/src/crypto/tls/conn.go:831 +0xd0
crypto/tls.(*Conn).readRecordOrCCS(0x14001506008, 0x0)
        /usr/local/go/src/crypto/tls/conn.go:629 +0x35c
crypto/tls.(*Conn).readRecord(...)
        /usr/local/go/src/crypto/tls/conn.go:591
crypto/tls.(*Conn).Read(0x14001506008, {0x14000fa2000, 0x8000, 0x1002c588c?})
        /usr/local/go/src/crypto/tls/conn.go:1385 +0x164
bufio.(*Reader).Read(0x14000be3440, {0x14000db84a0, 0x9, 0x1002e3460?})
        /usr/local/go/src/bufio/bufio.go:241 +0x1b4
io.ReadAtLeast({0x10316b800, 0x14000be3440}, {0x14000db84a0, 0x9, 0x9}, 0x9)
        /usr/local/go/src/io/io.go:335 +0xa0
io.ReadFull(...)
        /usr/local/go/src/io/io.go:354
golang.org/x/net/http2.readFrameHeader({0x14000db84a0, 0x9, 0x689bdedf0?}, {0x10316b800?, 0x14000be3440?})
        /Users/fatima/go/pkg/mod/golang.org/x/net@v0.26.0/http2/frame.go:237 +0x58
golang.org/x/net/http2.(*Framer).ReadFrame(0x14000db8460)
        /Users/fatima/go/pkg/mod/golang.org/x/net@v0.26.0/http2/frame.go:501 +0x78
google.golang.org/grpc/internal/transport.(*http2Client).reader(0x14000628908, 0x1400059b5e0)
        /Users/fatima/go/pkg/mod/google.golang.org/grpc@v1.62.1/internal/transport/http2_client.go:1593 +0x1a4
created by google.golang.org/grpc/internal/transport.newHTTP2Client in goroutine 462
        /Users/fatima/go/pkg/mod/google.golang.org/grpc@v1.62.1/internal/transport/http2_client.go:400 +0x14bc

goroutine 31 [IO wait]:
internal/poll.runtime_pollWait(0x12ba794b8, 0x72)
        /usr/local/go/src/runtime/netpoll.go:351 +0xa0
internal/poll.(*pollDesc).wait(0x14000c34100?, 0x14001496500?, 0x0)
        /usr/local/go/src/internal/poll/fd_poll_runtime.go:84 +0x28
internal/poll.(*pollDesc).waitRead(...)
        /usr/local/go/src/internal/poll/fd_poll_runtime.go:89
internal/poll.(*FD).Read(0x14000c34100, {0x14001496500, 0x500, 0x500})
        /usr/local/go/src/internal/poll/fd_unix.go:165 +0x1fc
net.(*netFD).Read(0x14000c34100, {0x14001496500?, 0x14001496500?, 0x5?})
        /usr/local/go/src/net/fd_posix.go:55 +0x28
net.(*conn).Read(0x140001aa0f0, {0x14001496500?, 0x14000c6a1c8?, 0x8?})
        /usr/local/go/src/net/net.go:189 +0x34
crypto/tls.(*atLeastReader).Read(0x14000ae8750, {0x14001496500?, 0x0?, 0x14000ae8750?})
        /usr/local/go/src/crypto/tls/conn.go:809 +0x40
bytes.(*Buffer).ReadFrom(0x14000c6a2b8, {0x10316bbe0, 0x14000ae8750})
        /usr/local/go/src/bytes/buffer.go:211 +0x90
crypto/tls.(*Conn).readFromUntil(0x14000c6a008, {0x104dae7e0, 0x140001aa0f0}, 0x0?)
        /usr/local/go/src/crypto/tls/conn.go:831 +0xd0
crypto/tls.(*Conn).readRecordOrCCS(0x14000c6a008, 0x0)
        /usr/local/go/src/crypto/tls/conn.go:629 +0x35c
crypto/tls.(*Conn).readRecord(...)
        /usr/local/go/src/crypto/tls/conn.go:591
crypto/tls.(*Conn).Read(0x14000c6a008, {0x14000db8200, 0x9, 0x1007ca860?})
        /usr/local/go/src/crypto/tls/conn.go:1385 +0x164
io.ReadAtLeast({0x104daea60, 0x14000c6a008}, {0x14000db8200, 0x9, 0x9}, 0x9)
        /usr/local/go/src/io/io.go:335 +0xa0
io.ReadFull(...)
        /usr/local/go/src/io/io.go:354
golang.org/x/net/http2.readFrameHeader({0x14000db8200, 0x9, 0x14000b7da40?}, {0x104daea60?, 0x14000c6a008?})
        /Users/fatima/go/pkg/mod/golang.org/x/net@v0.26.0/http2/frame.go:237 +0x58
golang.org/x/net/http2.(*Framer).ReadFrame(0x14000db81c0)
        /Users/fatima/go/pkg/mod/golang.org/x/net@v0.26.0/http2/frame.go:501 +0x78
golang.org/x/net/http2.(*serverConn).readFrames(0x140015441c0)
        /Users/fatima/go/pkg/mod/golang.org/x/net@v0.26.0/http2/server.go:858 +0x98
created by golang.org/x/net/http2.(*serverConn).serve in goroutine 563
        /Users/fatima/go/pkg/mod/golang.org/x/net@v0.26.0/http2/server.go:971 +0x460

goroutine 244 [chan receive]:
github.com/openbao/openbao/vault.(*Core).runStandby.func1()
        /Users/fatima/openbao/vault/ha.go:404 +0x24
github.com/oklog/run.(*Group).Run.func1({0x14000c26190?, 0x10315a048?})
        /Users/fatima/go/pkg/mod/github.com/oklog/run@v1.1.0/group.go:38 +0x34
created by github.com/oklog/run.(*Group).Run in goroutine 470
        /Users/fatima/go/pkg/mod/github.com/oklog/run@v1.1.0/group.go:37 +0x68

goroutine 104 [chan receive]:
github.com/gammazero/workerpool.worker(0x0?, 0x14000b6ea10, 0x14000c02030)
        /Users/fatima/go/pkg/mod/github.com/gammazero/workerpool@v1.1.3/workerpool.go:238 +0x40
created by github.com/gammazero/workerpool.(*WorkerPool).dispatch in goroutine 291
        /Users/fatima/go/pkg/mod/github.com/gammazero/workerpool@v1.1.3/workerpool.go:197 +0x214

goroutine 112 [select]:
google.golang.org/grpc/internal/grpcsync.(*CallbackSerializer).run(0x140010975e0, {0x1031938d8, 0x1400109b950})
        /Users/fatima/go/pkg/mod/google.golang.org/grpc@v1.62.1/internal/grpcsync/callback_serializer.go:76 +0xc4
created by google.golang.org/grpc/internal/grpcsync.NewCallbackSerializer in goroutine 8
        /Users/fatima/go/pkg/mod/google.golang.org/grpc@v1.62.1/internal/grpcsync/callback_serializer.go:52 +0x118

goroutine 510 [IO wait]:
internal/poll.runtime_pollWait(0x12ba798d8, 0x72)
        /usr/local/go/src/runtime/netpoll.go:351 +0xa0
internal/poll.(*pollDesc).wait(0x14000a23a00?, 0x14000a0c000?, 0x0)
        /usr/local/go/src/internal/poll/fd_poll_runtime.go:84 +0x28
internal/poll.(*pollDesc).waitRead(...)
        /usr/local/go/src/internal/poll/fd_poll_runtime.go:89
internal/poll.(*FD).Read(0x14000a23a00, {0x14000a0c000, 0xa80, 0xa80})
        /usr/local/go/src/internal/poll/fd_unix.go:165 +0x1fc
net.(*netFD).Read(0x14000a23a00, {0x14000a0c000?, 0x14000a0c028?, 0x5?})
        /usr/local/go/src/net/fd_posix.go:55 +0x28
net.(*conn).Read(0x14000a240b0, {0x14000a0c000?, 0x14000a74548?, 0x8?})
        /usr/local/go/src/net/net.go:189 +0x34
crypto/tls.(*atLeastReader).Read(0x14001514390, {0x14000a0c000?, 0x0?, 0x14001514390?})
        /usr/local/go/src/crypto/tls/conn.go:809 +0x40
bytes.(*Buffer).ReadFrom(0x14000a74638, {0x10316bbe0, 0x14001514390})
        /usr/local/go/src/bytes/buffer.go:211 +0x90
crypto/tls.(*Conn).readFromUntil(0x14000a74388, {0x104dae7e0, 0x14000a240b0}, 0x0?)
        /usr/local/go/src/crypto/tls/conn.go:831 +0xd0
crypto/tls.(*Conn).readRecordOrCCS(0x14000a74388, 0x0)
        /usr/local/go/src/crypto/tls/conn.go:629 +0x35c
crypto/tls.(*Conn).readRecord(...)
        /usr/local/go/src/crypto/tls/conn.go:591
crypto/tls.(*Conn).Read(0x14000a74388, {0x140014a1000, 0x1000, 0x1007e4628?})
        /usr/local/go/src/crypto/tls/conn.go:1385 +0x164
bufio.(*Reader).Read(0x140014936e0, {0x140006fad60, 0x9, 0x104859780?})
        /usr/local/go/src/bufio/bufio.go:241 +0x1b4
io.ReadAtLeast({0x10316b800, 0x140014936e0}, {0x140006fad60, 0x9, 0x9}, 0x9)
        /usr/local/go/src/io/io.go:335 +0xa0
io.ReadFull(...)
        /usr/local/go/src/io/io.go:354
golang.org/x/net/http2.readFrameHeader({0x140006fad60, 0x9, 0xfeb800?}, {0x10316b800?, 0x140014936e0?})
        /Users/fatima/go/pkg/mod/golang.org/x/net@v0.26.0/http2/frame.go:237 +0x58
golang.org/x/net/http2.(*Framer).ReadFrame(0x140006fad20)
        /Users/fatima/go/pkg/mod/golang.org/x/net@v0.26.0/http2/frame.go:501 +0x78
golang.org/x/net/http2.(*clientConnReadLoop).run(0x1400099ff98)
        /Users/fatima/go/pkg/mod/golang.org/x/net@v0.26.0/http2/transport.go:2358 +0xd0
golang.org/x/net/http2.(*ClientConn).readLoop(0x1400149e000)
        /Users/fatima/go/pkg/mod/golang.org/x/net@v0.26.0/http2/transport.go:2254 +0x78
created by golang.org/x/net/http2.(*Transport).newClientConn in goroutine 494
        /Users/fatima/go/pkg/mod/golang.org/x/net@v0.26.0/http2/transport.go:869 +0xae0

goroutine 246 [select]:
github.com/openbao/openbao/vault.(*Core).periodicLeaderRefresh(0x14000b97508, 0x0, 0x14000baeaf0)
        /Users/fatima/openbao/vault/ha.go:809 +0x8c
github.com/openbao/openbao/vault.(*Core).runStandby.func5()
        /Users/fatima/openbao/vault/ha.go:425 +0x28
github.com/oklog/run.(*Group).Run.func1({0x14001004300?, 0x140008fe210?})
        /Users/fatima/go/pkg/mod/github.com/oklog/run@v1.1.0/group.go:38 +0x34
created by github.com/oklog/run.(*Group).Run in goroutine 470
        /Users/fatima/go/pkg/mod/github.com/oklog/run@v1.1.0/group.go:37 +0x68

goroutine 113 [select]:
google.golang.org/grpc/internal/grpcsync.(*CallbackSerializer).run(0x14001097610, {0x1031938d8, 0x1400109b9a0})
        /Users/fatima/go/pkg/mod/google.golang.org/grpc@v1.62.1/internal/grpcsync/callback_serializer.go:76 +0xc4
created by google.golang.org/grpc/internal/grpcsync.NewCallbackSerializer in goroutine 8
        /Users/fatima/go/pkg/mod/google.golang.org/grpc@v1.62.1/internal/grpcsync/callback_serializer.go:52 +0x118

goroutine 307 [select]:
github.com/openbao/openbao/vault.(*Core).setupCachedMFAResponseAuth.func1()
        /Users/fatima/openbao/vault/core.go:3243 +0x98
created by github.com/openbao/openbao/vault.(*Core).setupCachedMFAResponseAuth in goroutine 236
        /Users/fatima/openbao/vault/core.go:3240 +0x180

goroutine 308 [select]:
github.com/openbao/openbao/vault.(*Core).metricsLoop(0x14000b96008, 0x14000b6eee0)
        /Users/fatima/openbao/vault/core_metrics.go:50 +0x118
github.com/openbao/openbao/vault.(*Core).emitMetricsActiveNode(0x14000b96008, 0x14000b6eee0)
        /Users/fatima/openbao/vault/core_metrics.go:261 +0x76c
created by github.com/openbao/openbao/vault.standardUnsealStrategy.unseal in goroutine 236
        /Users/fatima/openbao/vault/core.go:2308 +0x578

goroutine 502 [chan receive]:
github.com/gammazero/workerpool.worker(0x0?, 0x14000b6ea10, 0x14000c02030)
        /Users/fatima/go/pkg/mod/github.com/gammazero/workerpool@v1.1.3/workerpool.go:238 +0x40
created by github.com/gammazero/workerpool.(*WorkerPool).dispatch in goroutine 291
        /Users/fatima/go/pkg/mod/github.com/gammazero/workerpool@v1.1.3/workerpool.go:197 +0x214

goroutine 481 [chan receive]:
github.com/gammazero/workerpool.worker(0x0?, 0x14000b6ea10, 0x14000c02030)
        /Users/fatima/go/pkg/mod/github.com/gammazero/workerpool@v1.1.3/workerpool.go:238 +0x40
created by github.com/gammazero/workerpool.(*WorkerPool).dispatch in goroutine 291
        /Users/fatima/go/pkg/mod/github.com/gammazero/workerpool@v1.1.3/workerpool.go:197 +0x214

goroutine 470 [chan receive]:
github.com/oklog/run.(*Group).Run(0x140008fe1c8)
        /Users/fatima/go/pkg/mod/github.com/oklog/run@v1.1.0/group.go:43 +0x140
github.com/openbao/openbao/vault.(*Core).runStandby(0x14000b97508, 0x14000b6ef50?, 0x14000d97730, 0x14000d977a0)
        /Users/fatima/openbao/vault/ha.go:457 +0x778
created by github.com/openbao/openbao/vault.(*Core).unsealInternal in goroutine 8
        /Users/fatima/openbao/vault/core.go:1867 +0x1f8

goroutine 562 [select]:
github.com/openbao/openbao/vault.(*requestForwardingHandler).Handoff.func1()
        /Users/fatima/openbao/vault/request_forwarding.go:158 +0x70
created by github.com/openbao/openbao/vault.(*requestForwardingHandler).Handoff in goroutine 282
        /Users/fatima/openbao/vault/request_forwarding.go:157 +0xfc

goroutine 477 [select]:
github.com/openbao/openbao/sdk/v2/physical/inmem.(*InmemLock).Lock(0x14000da4f80, 0x14000d97b20)
        /Users/fatima/openbao/sdk/physical/inmem/inmem_ha.go:107 +0x188
github.com/openbao/openbao/vault.(*Core).acquireLock(0x14000b98a08, {0x1031858c0, 0x14000da4f80}, 0x14000d97b20)
        /Users/fatima/openbao/vault/ha.go:1053 +0x48
github.com/openbao/openbao/vault.(*Core).waitForLeadership(0x14000b98a08, 0x0?, 0x14000d978f0, 0x14000d97b20)
        /Users/fatima/openbao/vault/ha.go:499 +0x248
github.com/openbao/openbao/vault.(*Core).runStandby.func9()
        /Users/fatima/openbao/vault/ha.go:448 +0x2c
github.com/oklog/run.(*Group).Run.func1({0x14000e14cf0?, 0x14000980a38?})
        /Users/fatima/go/pkg/mod/github.com/oklog/run@v1.1.0/group.go:38 +0x34
created by github.com/oklog/run.(*Group).Run in goroutine 472
        /Users/fatima/go/pkg/mod/github.com/oklog/run@v1.1.0/group.go:37 +0x68

goroutine 478 [sync.Cond.Wait]:
sync.runtime_notifyListWait(0x140001bb550, 0x1)
        /usr/local/go/src/runtime/sema.go:587 +0x154
sync.(*Cond).Wait(0x140001bb540)
        /usr/local/go/src/sync/cond.go:71 +0xcc
github.com/openbao/openbao/sdk/v2/physical/inmem.(*InmemLock).Lock.func1()
        /Users/fatima/openbao/sdk/physical/inmem/inmem_ha.go:87 +0xc8
created by github.com/openbao/openbao/sdk/v2/physical/inmem.(*InmemLock).Lock in goroutine 477
        /Users/fatima/openbao/sdk/physical/inmem/inmem_ha.go:82 +0x154

goroutine 473 [chan receive]:
github.com/openbao/openbao/vault.(*Core).runStandby.func1()
        /Users/fatima/openbao/vault/ha.go:404 +0x24
github.com/oklog/run.(*Group).Run.func1({0x14000d83900?, 0x10315a048?})
        /Users/fatima/go/pkg/mod/github.com/oklog/run@v1.1.0/group.go:38 +0x34
created by github.com/oklog/run.(*Group).Run in goroutine 472
        /Users/fatima/go/pkg/mod/github.com/oklog/run@v1.1.0/group.go:37 +0x68

goroutine 475 [select]:
github.com/openbao/openbao/vault.(*Core).periodicLeaderRefresh(0x14000b98a08, 0x0, 0x14000d97a40)
        /Users/fatima/openbao/vault/ha.go:809 +0x8c
github.com/openbao/openbao/vault.(*Core).runStandby.func5()
        /Users/fatima/openbao/vault/ha.go:425 +0x28
github.com/oklog/run.(*Group).Run.func1({0x14000be55e0?, 0x140009809a8?})
        /Users/fatima/go/pkg/mod/github.com/oklog/run@v1.1.0/group.go:38 +0x34
created by github.com/oklog/run.(*Group).Run in goroutine 472
        /Users/fatima/go/pkg/mod/github.com/oklog/run@v1.1.0/group.go:37 +0x68

goroutine 469 [IO wait]:
internal/poll.runtime_pollWait(0x12ba79ae8, 0x72)
        /usr/local/go/src/runtime/netpoll.go:351 +0xa0
internal/poll.(*pollDesc).wait(0x14000dd1a80?, 0x1002c6650?, 0x0)
        /usr/local/go/src/internal/poll/fd_poll_runtime.go:84 +0x28
internal/poll.(*pollDesc).waitRead(...)
        /usr/local/go/src/internal/poll/fd_poll_runtime.go:89
internal/poll.(*FD).Accept(0x14000dd1a80)
        /usr/local/go/src/internal/poll/fd_unix.go:620 +0x24c
net.(*netFD).accept(0x14000dd1a80)
        /usr/local/go/src/net/fd_unix.go:172 +0x28
net.(*TCPListener).accept(0x14000da4dc0)
        /usr/local/go/src/net/tcpsock_posix.go:159 +0x28
net.(*TCPListener).Accept(0x14000da4dc0)
        /usr/local/go/src/net/tcpsock.go:372 +0x2c
crypto/tls.(*listener).Accept(0x14000980720)
        /usr/local/go/src/crypto/tls/tls.go:67 +0x30
github.com/openbao/openbao/vault/cluster.(*Listener).Run.func1(0x14000d97650, {0x103184550, 0x14000980720})
        /Users/fatima/openbao/vault/cluster/cluster.go:328 +0x10c
created by github.com/openbao/openbao/vault/cluster.(*Listener).Run in goroutine 8
        /Users/fatima/openbao/vault/cluster/cluster.go:301 +0x27c

goroutine 472 [chan receive]:
github.com/oklog/run.(*Group).Run(0x14000980918)
        /Users/fatima/go/pkg/mod/github.com/oklog/run@v1.1.0/group.go:43 +0x140
github.com/openbao/openbao/vault.(*Core).runStandby(0x14000b98a08, 0x14000b6ef50?, 0x14000d978f0, 0x14000d97960)
        /Users/fatima/openbao/vault/ha.go:457 +0x778
created by github.com/openbao/openbao/vault.(*Core).unsealInternal in goroutine 8
        /Users/fatima/openbao/vault/core.go:1867 +0x1f8

goroutine 474 [select]:
github.com/openbao/openbao/vault.(*Core).periodicCheckKeyUpgrades(0x14000b98a08, {0x1031937c0, 0x1049b79e0}, 0x14000d979d0)
        /Users/fatima/openbao/vault/ha.go:868 +0xd8
github.com/openbao/openbao/vault.(*Core).runStandby.func3()
        /Users/fatima/openbao/vault/ha.go:413 +0x34
github.com/oklog/run.(*Group).Run.func1({0x14000980948?, 0x14000980978?})
        /Users/fatima/go/pkg/mod/github.com/oklog/run@v1.1.0/group.go:38 +0x34
created by github.com/oklog/run.(*Group).Run in goroutine 472
        /Users/fatima/go/pkg/mod/github.com/oklog/run@v1.1.0/group.go:37 +0x68

goroutine 476 [select]:
github.com/openbao/openbao/vault.(*Core).metricsLoop(0x14000b98a08, 0x14000d97ab0)
        /Users/fatima/openbao/vault/core_metrics.go:50 +0x118
github.com/openbao/openbao/vault.(*Core).runStandby.func7()
        /Users/fatima/openbao/vault/ha.go:436 +0x24
github.com/oklog/run.(*Group).Run.func1({0x140009809d8?, 0x14000980a08?})
        /Users/fatima/go/pkg/mod/github.com/oklog/run@v1.1.0/group.go:38 +0x34
created by github.com/oklog/run.(*Group).Run in goroutine 472
        /Users/fatima/go/pkg/mod/github.com/oklog/run@v1.1.0/group.go:37 +0x68

goroutine 463 [select]:
github.com/openbao/openbao/vault.(*forwardingClient).startHeartbeat.func1()
        /Users/fatima/openbao/vault/request_forwarding_rpc.go:171 +0x1dc
created by github.com/openbao/openbao/vault.(*forwardingClient).startHeartbeat in goroutine 8
        /Users/fatima/openbao/vault/request_forwarding_rpc.go:119 +0x5c

goroutine 465 [IO wait]:
internal/poll.runtime_pollWait(0x12ba79f08, 0x72)
        /usr/local/go/src/runtime/netpoll.go:351 +0xa0
internal/poll.(*pollDesc).wait(0x140006ea500?, 0x1002d88d8?, 0x0)
        /usr/local/go/src/internal/poll/fd_poll_runtime.go:84 +0x28
internal/poll.(*pollDesc).waitRead(...)
        /usr/local/go/src/internal/poll/fd_poll_runtime.go:89
internal/poll.(*FD).Accept(0x140006ea500)
        /usr/local/go/src/internal/poll/fd_unix.go:620 +0x24c
net.(*netFD).accept(0x140006ea500)
        /usr/local/go/src/net/fd_unix.go:172 +0x28
net.(*TCPListener).accept(0x140001bb200)
        /usr/local/go/src/net/tcpsock_posix.go:159 +0x28
net.(*TCPListener).Accept(0x140001bb200)
        /usr/local/go/src/net/tcpsock.go:372 +0x2c
crypto/tls.(*listener).Accept(0x140008fe120)
        /usr/local/go/src/crypto/tls/tls.go:67 +0x30
net/http.(*Server).Serve(0x140003a8e10, {0x103184c60, 0x140008fe138})
        /usr/local/go/src/net/http/server.go:3330 +0x294
created by github.com/openbao/openbao/vault.(*TestCluster).start in goroutine 8
        /Users/fatima/openbao/vault/testing.go:835 +0x3a8

goroutine 530 [IO wait]:
internal/poll.runtime_pollWait(0x12ba79e00, 0x72)
        /usr/local/go/src/runtime/netpoll.go:351 +0xa0
internal/poll.(*pollDesc).wait(0x140006ea780?, 0x14000aea500?, 0x0)
        /usr/local/go/src/internal/poll/fd_poll_runtime.go:84 +0x28
internal/poll.(*pollDesc).waitRead(...)
        /usr/local/go/src/internal/poll/fd_poll_runtime.go:89
internal/poll.(*FD).Accept(0x140006ea780)
        /usr/local/go/src/internal/poll/fd_unix.go:620 +0x24c
net.(*netFD).accept(0x140006ea780)
        /usr/local/go/src/net/fd_unix.go:172 +0x28
net.(*TCPListener).accept(0x140001bb300)
        /usr/local/go/src/net/tcpsock_posix.go:159 +0x28
net.(*TCPListener).Accept(0x140001bb300)
        /usr/local/go/src/net/tcpsock.go:372 +0x2c
crypto/tls.(*listener).Accept(0x140008fe330)
        /usr/local/go/src/crypto/tls/tls.go:67 +0x30
net/http.(*Server).Serve(0x140003a8f00, {0x103184c60, 0x140008fe348})
        /usr/local/go/src/net/http/server.go:3330 +0x294
created by github.com/openbao/openbao/vault.(*TestCluster).start in goroutine 8
        /Users/fatima/openbao/vault/testing.go:835 +0x3a8

goroutine 531 [IO wait]:
internal/poll.runtime_pollWait(0x12ba79cf8, 0x72)
        /usr/local/go/src/runtime/netpoll.go:351 +0xa0
internal/poll.(*pollDesc).wait(0x140006eaa00?, 0x14000aea550?, 0x0)
        /usr/local/go/src/internal/poll/fd_poll_runtime.go:84 +0x28
internal/poll.(*pollDesc).waitRead(...)
        /usr/local/go/src/internal/poll/fd_poll_runtime.go:89
internal/poll.(*FD).Accept(0x140006eaa00)
        /usr/local/go/src/internal/poll/fd_unix.go:620 +0x24c
net.(*netFD).accept(0x140006eaa00)
        /usr/local/go/src/net/fd_unix.go:172 +0x28
net.(*TCPListener).accept(0x140001bb3c0)
        /usr/local/go/src/net/tcpsock_posix.go:159 +0x28
net.(*TCPListener).Accept(0x140001bb3c0)
        /usr/local/go/src/net/tcpsock.go:372 +0x2c
crypto/tls.(*listener).Accept(0x140008fe528)
        /usr/local/go/src/crypto/tls/tls.go:67 +0x30
net/http.(*Server).Serve(0x140003a8ff0, {0x103184c60, 0x140008fe540})
        /usr/local/go/src/net/http/server.go:3330 +0x294
created by github.com/openbao/openbao/vault.(*TestCluster).start in goroutine 8
        /Users/fatima/openbao/vault/testing.go:835 +0x3a8

goroutine 534 [select]:
net/http.(*http2serverConn).serve(0x14001544000)
        /usr/local/go/src/net/http/h2_bundle.go:4812 +0x6ec
net/http.(*http2Server).serveConn(0x14000ba8960, {0x10319c5b8, 0x14001279c08}, 0x1400142db28, 0x0)
        /usr/local/go/src/net/http/h2_bundle.go:4402 +0x97c
net/http.(*http2Server).ServeConn(...)
        /usr/local/go/src/net/http/h2_bundle.go:4266
net/http.http2ConfigureServer.func1(0x140003a8e10, 0x14001279c08, {0x10316c5c0, 0x14001518100})
        /usr/local/go/src/net/http/h2_bundle.go:4185 +0x100
net/http.(*conn).serve(0x14001414a20, {0x1031938a0, 0x1400127ef60})
        /usr/local/go/src/net/http/server.go:2000 +0xfa0
created by net/http.(*Server).Serve in goroutine 465
        /usr/local/go/src/net/http/server.go:3360 +0x3dc

goroutine 598 [select]:
google.golang.org/grpc/internal/transport.(*controlBuffer).get(0x14000ae4be0, 0x1)
        /Users/fatima/go/pkg/mod/google.golang.org/grpc@v1.62.1/internal/transport/controlbuf.go:418 +0x14c
google.golang.org/grpc/internal/transport.(*loopyWriter).run(0x1400059b650)
        /Users/fatima/go/pkg/mod/google.golang.org/grpc@v1.62.1/internal/transport/controlbuf.go:551 +0x74
google.golang.org/grpc/internal/transport.newHTTP2Client.func6()
        /Users/fatima/go/pkg/mod/google.golang.org/grpc@v1.62.1/internal/transport/http2_client.go:454 +0x70
created by google.golang.org/grpc/internal/transport.newHTTP2Client in goroutine 462
        /Users/fatima/go/pkg/mod/google.golang.org/grpc@v1.62.1/internal/transport/http2_client.go:452 +0x190c
FAIL    github.com/openbao/openbao/builtin/logical/pki  30.956s
```
</details>